### PR TITLE
Use new m1 mac runners for CI

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Updates the CI runner to [use M1 macs](https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/). This should be faster than the existing runner & is free for public repos.

### CODE REVIEW

File: .github/workflows/functional-tests.yml [hunk 1]
```diff
@@ -16,12 +16,12 @@ env:
...
   test:
-    runs-on: macos-latest
+    runs-on: macos-14
     timeout-minutes: 45
     strategy:
       matrix:
         jdk-version: ["17"]
-        ndk-version: ["21.4.7075529"]
+#        ndk-version: ["21.4.7075529"]
         api-level: [29]
```
Comment (bidetofevil) [inline .github/workflows/functional-tests.yml:25]: Not for this PR, but should be bump this to the latest? And also test on the minimum?

**Review by bidetofevil — APPROVED** [review]
LGTM

File: .github/workflows/functional-tests.yml (hunk 1 above)
Comment (fractalwrench) [reply .github/workflows/functional-tests.yml:24]: Emulators beyond API 29 aren't supported by the GH action we use for running emulators yet
